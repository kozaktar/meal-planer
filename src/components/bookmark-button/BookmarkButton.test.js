import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import BookmarkButton from './BookmarkButton.component';
 
const mockStore = configureStore([]);

describe('My Connected React-Redux Component', () => {
    let store;
    let component;
    let mockProps;
   
    beforeEach(() => {
        store = mockStore({
          user:{currentUser:{authID:'12345'}},
          recipes:{recipes:{savingRecipe:false}}
        });

        mockProps={
            classes:{bookmark:null, icon:null},
            recipe:{
                _id:'a123',
                users:['54355', '12345']
        }
    }
     
        component = renderer.create(
          <Provider store={store}>
            <BookmarkButton {...mockProps}/>
          </Provider>
        );
      });
   
    it('Bookmark button should render without crashing', () => {
        expect(component).toMatchSnapshot()
    });
   
    it('should dispatch an action on button click', () => {
   
    });
  });