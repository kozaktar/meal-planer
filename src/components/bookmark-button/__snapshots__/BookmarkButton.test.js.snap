// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmark button component should render the bookmark button 1`] = `undefined`;

exports[`My Connected React-Redux Component Bookmark button should render without crashing 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root"
  disabled={false}
  id="bookmarkButton"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-describedby={null}
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
      focusable="false"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      title="Remove recipe"
      viewBox="0 0 24 24"
    >
      <path
        d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2z"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
